dpnp/dpnp_iface_indexing.py

```diff
@@ -117,50 +120,176 @@ def _ravel_multi_index_checks(multi_index, dims, order):
         )
 
 
-def choose(x1, choices, out=None, mode="raise"):
+def _build_choices_list(choices):
+    """
+    Gather queues and USM types for the input, expected to be an array or
+    list of arrays. If a single array of dimension greater than one, the array
+    will be unstacked.
+
+    Returns a list of :class:`dpctl.tensor.usm_ndarray`s.
+    """
+
+    if dpnp.is_supported_array_type(choices):
+        choices = dpnp.unstack(choices)
+    elif not isinstance(choices, Iterable):
+        raise TypeError("`choices` must be an array or sequence of arrays")
+    return [dpnp.get_usm_ndarray(chc) for chc in choices]
+
+
+def _choose_run(inds, chcs, q, usm_type, out=None, mode=0):
+    # arg validation, broadcasting, type coercion assumed done by caller
+    if out is not None:
+        out = dpnp.get_usm_ndarray(out)
+
+        if not out.flags.writable:
+            raise ValueError("provided `out` array is read-only")
+
+        if out.shape != inds.shape:
+            raise ValueError(
+                "The shape of input and output arrays are inconsistent. "
+                f"Expected output shape is {inds.shape}, got {out.shape}"
+            )
+
+        if chcs[0].dtype != out.dtype:
+            raise TypeError(
+                f"Output array of type {chcs[0].dtype} is needed, "
+                f"got {out.dtype}"
+            )
+
+        if dpu.get_execution_queue((q, out.sycl_queue)) is None:
+            raise dpu.ExecutionPlacementError(
+                "Input and output allocation queues are not compatible"
+            )
+
+        if ti._array_overlap(inds, out) or any(
+            ti._array_overlap(out, chc) for chc in chcs
+        ):
+            # Allocate a temporary buffer to avoid memory overlapping.
+            out = dpt.empty_like(out)
+    else:
+        out = dpt.empty(
+            inds.shape, dtype=chcs[0].dtype, usm_type=usm_type, sycl_queue=q
+        )
+
+    _manager = dpu.SequentialOrderManager[q]
+    dep_evs = _manager.submitted_events
+
+    h_ev, choose_ev = indexing_ext._choose(inds, chcs, out, mode, q, dep_evs)
+    _manager.add_event_pair(h_ev, choose_ev)
+
+    return out
+
+
+def choose(a, choices, out=None, mode="wrap"):
     """
     Construct an array from an index array and a set of arrays to choose from.
 
     For full documentation refer to :obj:`numpy.choose`.
 
+    Parameters
+    ----------
+    a : {dpnp.ndarray, usm_ndarray}
+        An integer array of indices indicating the position of the array
+        in `choices` to choose from. Behavior of out-of-bounds integers (i.e.,
+        integers outside of `[0, n-1]` where `n` is the number of choices) is
+        determined by the `mode` keyword.
+    choices : {dpnp.ndarray, usm_ndarray,
+               sequence of dpnp.ndarrays and usm_ndarrays}
```

Comment:
I've build documentation (I run `make html` in `dpnp/doc` folder to render the documentation) and it seems it's still generated in a wrong way:

![image](https://github.com/user-attachments/assets/75d74649-068b-4b95-ac73-cbc664056033)

JFY, in future, it would be better to avoid creating a PR as a push from the forked repo, because in that case it seems impossible for GitHub action to render the documentation automatically. And so forces manual steps to render.

Reply:
adding a backslash should resolve the issue here.

```suggestion
    choices : {dpnp.ndarray, usm_ndarray, \
               sequence of dpnp.ndarrays and usm_ndarrays}
```